exif2timestream.py

```diff
@@ -236,18 +249,111 @@ def __call__(self, x):
         return None
 
 
+def resize_img(filename, to_width):
+    # Open the Image and get its width
+    if not(SKIMAGE):
+        warnings.warn(
+            "Skimage Not Installed, Unable to Test Resize", ImportWarning)
+        return None
+    img = io.imread(filename)
+    w = novice.open(filename).width
+    scale = float(to_width) / w
+    # Rescale the image
+    img = rescale(img, scale)
+    # read in old exxif data
+    exif_source = pexif.JpegFile.fromFile(filename)
+    # Save image
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        io.imsave(filename, img)
+    # Write new exif data from old image
+    try:
+        exif_dest = pexif.JpegFile.fromFile(filename)
+        exif_dest.exif.primary.ExtendedEXIF.DateTimeOriginal = exif_source.exif.primary.ExtendedEXIF.DateTimeOriginal
+        exif_dest.writeFile(filename)
+    except AttributeError:
+        pass
+
+
+def get_time_from_filename(filename, mask=EXIF_DATE_MASK):
+    # Replace the year with the regex equivalent to parse
+    regex_mask = mask.replace("%Y", "\d{4}").replace(
+        "%m", "\d{2}").replace("%d", "\d{2}")
+    regex_mask = regex_mask.replace("%H", "\d{2}").replace(
+        "%M", "\d{2}").replace("%S", "\d{2}")
+    # Wildcard character before and after the regex
+    regex_mask = "\.*" + regex_mask + "\.*"
+    # compile the regex
+    date_reg_exp = re.compile(regex_mask)
+    # get the list of possible date matches
+    matches_list = date_reg_exp.findall(filename)
+    for match in matches_list:
+        try:
+            # Parse each match into a datetime
+            datetime = strptime(match, mask)
+            # Return the datetime
+            return datetime
+        # If we cant convert it to the date, then go to the next item on the
+        # list
+        except ValueError:
+            continue
+    # If we cant match anything, then return null
```

Comment:
Don't worry about it, but the comment mentions null, and that's not python. Just pedandtry :)